pkg/controller/tiers/tiers_controller.go

```diff
@@ -0,0 +1,165 @@
+
+	c, err := controller.New("tiers-controller", mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Error(err, "Failed to establish a connection to k8s")
+		return err
+	}
+
+	go utils.WaitToAddTierWatch(networkpolicy.TigeraComponentTierName, c, k8sClient, log, tierWatchReady)
+
+	go utils.WaitToAddNetworkPolicyWatches(c, k8sClient, log, policyWatchesReady, []types.NamespacedName{
+		{Name: tiers.ClusterDNSPolicyName, Namespace: "openshift-dns"},
+		{Name: tiers.ClusterDNSPolicyName, Namespace: "kube-system"},
+	})
+
+	return add(mgr, c)
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, opts options.AddOptions, tierWatchReady *utils.ReadyFlag, policyWatchesReady *utils.ReadyFlag) reconcile.Reconciler {
+	r := &ReconcileTiers{
+		client:             mgr.GetClient(),
+		scheme:             mgr.GetScheme(),
+		provider:           opts.DetectedProvider,
+		status:             status.New(mgr.GetClient(), "tiers", opts.KubernetesVersion),
+		tierWatchReady:     tierWatchReady,
+		policyWatchesReady: policyWatchesReady,
+	}
+	r.status.Run(opts.ShutdownContext)
+	return r
+}
+
+// add adds watches for resources that are available at startup.
+func add(mgr manager.Manager, c controller.Controller) error {
+	if err := utils.AddNetworkWatch(c); err != nil {
+		return fmt.Errorf("tiers-controller failed to watch Tigera network resource: %v", err)
+	}
+
+	if err := utils.AddAPIServerWatch(c); err != nil {
+		return fmt.Errorf("tiers-controller failed to watch APIServer resource: %v", err)
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileTiers{}
+
+type ReconcileTiers struct {
+	client             client.Client
+	scheme             *runtime.Scheme
+	provider           operatorv1.Provider
+	status             status.StatusManager
+	tierWatchReady     *utils.ReadyFlag
+	policyWatchesReady *utils.ReadyFlag
+}
+
+func (r *ReconcileTiers) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Tiers")
+
+	if !utils.IsAPIServerReady(r.client, reqLogger) {
+		r.status.SetDegraded("Waiting for Tigera API server to be ready", "")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	if !r.tierWatchReady.IsReady() {
+		r.status.SetDegraded("Waiting for Tier watch to be established", "")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	if !r.policyWatchesReady.IsReady() {
+		r.status.SetDegraded("Waiting for NetworkPolicy watches to be established", "")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Ensure that a license is present so this controller can establish/manage tiers and domain-based policy.
```

Comment:
Normally we use "domain-based policy" to refer to policies that use domain names in their rules, so this would be more clear if it said "tiers and policy to ensure cluster DNS is not broken"

Reply:
Nice catch, originally this comment was in reference to the Guardian policy which had the possibility of using a domain name in its egress rules. It's now out of place given the refactor.

Will address in current PR.